src/trim_comparison.rs

```diff
@@ -0,0 +1,808 @@
+use std::{
+    borrow::Borrow,
+    collections::{BTreeMap, HashMap},
+    fs::OpenOptions,
+    ops::Deref,
+    usize,
+};
+
+use crate::{
+    io::{self, block_num_from_path, path_for_overall_report, try_deserialize, try_serialize},
+    juno_manager::{ManagerError, Network},
+    trace_comparison::{parse_same_string, string_is_empty, string_is_same, value_is_same},
+    utils::{self, felt_to_hex},
+};
+
+use anyhow::Context;
+use futures::future;
+use glob::glob;
+use itertools::Itertools;
+use log::{debug, info, warn};
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
+use tokio::task::{self};
+
+use petgraph::{
+    graph::{Graph, Node, NodeIndex},
+    visit::Dfs,
+};
+
+use crate::trace_comparison::DIFFERENT;
+
+struct TrimConfig {
+    level: usize,
+}
+
+fn trim_comparison_file(
+    block_num: u64,
+    network: Network,
+    config: TrimConfig,
+) -> Result<(), anyhow::Error> {
+    let path = io::succesful_comparison_path(block_num, network);
+    let comparison_file = OpenOptions::new()
+        .read(true)
+        .open(path.clone())
+        .context(format!(
+            "Reading comparison file: '{}'",
+            path.to_str().expect("failed to unwrap path")
+        ))?;
+    let reader = std::io::BufReader::new(comparison_file);
+    let comparison: Value = serde_json::from_reader(reader)?;
+
+    let trimmed_comparison = trim_comparison(comparison, &config);
+
+    let out_path = io::trimmed_comparison_path(block_num, network);
+    let comparison_file = OpenOptions::new()
+        .write(true)
```

Comment:
Why is the write flag true

Reply:
Here it writes back out to `"results-{network}/comparison-{block_num}-trimmed.json"`